larpandoracontent/LArHelpers/LArDiscreteProbabilityHelper.h

```diff
@@ -0,0 +1,249 @@
+
+
+template <typename T>
+inline T LArDiscreteProbabilityHelper::MakeRandomisedSample(const T &t, std::mt19937 &randomNumberGenerator)
+{
+    return LArDiscreteProbabilityHelper::MakeRandomisedSampleImpl(t, randomNumberGenerator);
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+template <>
+inline DiscreteProbabilityVector LArDiscreteProbabilityHelper::MakeRandomisedSampleImpl(const DiscreteProbabilityVector &t, 
+    std::mt19937 &randomNumberGenerator)
+{
+    return DiscreteProbabilityVector(t,randomNumberGenerator);
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+template <typename T>
+inline std::vector<T> LArDiscreteProbabilityHelper::MakeRandomisedSampleImpl(const std::vector<T> &t, std::mt19937 &randomNumberGenerator)
+{
+    std::vector<T> randomisedVector(t);
+    std::shuffle(randomisedVector.begin(), randomisedVector.end(), randomNumberGenerator);
+
+    return randomisedVector;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+template <typename T>
+inline size_t LArDiscreteProbabilityHelper::GetSize(const T &t)
+{
+    return LArDiscreteProbabilityHelper::GetSizeImpl(t);
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+template <>
+inline size_t LArDiscreteProbabilityHelper::GetSizeImpl(const DiscreteProbabilityVector& t)
+{
+    return t.GetSize();
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+template <typename T>
+inline size_t LArDiscreteProbabilityHelper::GetSizeImpl(const std::vector<T> &t)
+{
+    return t.size();
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+template <typename T>
+inline float LArDiscreteProbabilityHelper::GetElement(const T &t, const size_t index)
+{
+    return LArDiscreteProbabilityHelper::GetElementImpl(t, index);
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+template <>
+inline float LArDiscreteProbabilityHelper::GetElementImpl(const DiscreteProbabilityVector& t, const size_t index)
+{
+    return static_cast<float>(t.GetProbability(index));
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+template<typename T>
+inline float LArDiscreteProbabilityHelper::GetElementImpl(const std::vector<T> &t, const size_t index)
+{
+    return static_cast<float>(t.at(index));
+}
+
+} // namespace lar_content
```

Comment:
Done

Reply:
Empty line was intended to be between the line ending the namespace and the #endif line, rather than adding two lines between the inline function implementation and the line ending the namespace 😀

Reply:
hahahaha I am the worst.
Fixed (again).